### PR TITLE
release-1.23: update branch cut deadlines

### PR DESCRIPTION
**Pull request by palnabarun**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

The release cut dates are updated to the following:

1.23.0-alpha.2 -> Week 4 // Tue September 14, 2021
1.23.0-alpha.3 -> Week 6 // Tue September 28, 2021
1.23.0-alpha.4 -> Week 11 // Tue November 2, 2021
1.23.0-beta.0 -> Week 13 // Wed November 17, 2021

Note: beta.0 is not cut on a Tuesday because on that week's Tuesday,
Code Freeze is going to begin. By cutting the branch a day later, we
include last-minute merges to be included in a release cut. This is
beneficial for testing by prerelease consumers as the next release
(rc.0) is a week away.

Ref: [Release Engineering Meeting Notes](https://bit.ly/k8s-releng-meeting)

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

/assign @kubernetes/sig-release-leads
/assign @reylejano
/area release-team
/priority important-soon

### CODE REVIEW

**Comment by justaugustus** [comment]
cc: @kubernetes/release-engineering 
/hold for reviews

**Comment by puerco** [comment]
#### A note about beta.0 and beta.1:

During 1.22 we crammed a second beta (beta.1) between code freeze and branch as we had that request from the community in 1.21. The idea is to have a final prerelease to test merges allowed through exceptions after code freeze but before the branch gets forked and the rc is cut. I think it is a good idea as it lets folks test their changes while still in beta.

Semantically, I think that gave some coherence to alphas, betas, and RCs. All of which have been changing meaning since the branch cut moved from beta to RC:
* alpha: Codebase still in motion. Before code freeze.
* beta: Branch is frozen. Rate of change slows down considerably as only exceptions are allowed.
* RC: First look at the shape of the final release. Only release blockers go in. 

As it stands now, a beta is really an alpha with another name. Could be really named alpha.4. Naming aside, the prerelease between  CF and branch cut is missing. Should we add it?

**Comment by palnabarun** [comment]
Following the definition put out by @puerco, I think beta.0 is misplaced right now.

@puerco -- Some context from the meeting which is not in the notes (but they're in the recording), the reason the cut around code freeze was offset by +1 day was because of the reason you mentioned above about letting folks test the changes which have been allowed through exceptions.

Having said that, I am +1 if we want to bring back alpha.4 on Week 11. beta.0 on Week 13 and either of the following:
1. squeeze a beta.1 between beta.0 on Week 13 and rc.0 on Week 14
2. push rc.0 to a week later and have beta.1 on Week 14

**Comment by reylejano** [comment]
My opinions below are based on previous comments from release managers or SIG Release chairs and tech leads:
- From @puerco 's [definition](https://github.com/kubernetes/sig-release/pull/1688#issuecomment-909430003) of beta where the branch is frozen and also @saschagrunert 's [comment aligning beta.0 with Code Freeze](https://github.com/kubernetes/sig-release/pull/1645#discussion_r680778044) on week 13 , I am a +1 to keep beta.0 on week 13 with Code Freeze and have an alpha.4 on week 11
- @saschagrunert proposed to couple rc.0 with Test Freeze in this [comment](https://github.com/kubernetes/sig-release/pull/1645#discussion_r680773711) in a PR for the 1.23 schedule
- With these suggestions in mind, I'm a +1 to squeezing a beta.1 between beta.0 on week 13 and keep rc.0 on week 14 coupled with Test Freeze

**Comment by saschagrunert** [comment]
My opinion aligns with the comments from @puerco, @palnabarun and @reylejano :+1:

**Comment by palnabarun** [comment]
@reylejano @puerco @saschagrunert -- I've made the changes to alpha and beta according to what we discussed above.

I have not added a beta.1 in between beta.0 and rc.0 because I feel they are already 6 days apart and cutting a release in between may be tedious. What do you all think about it?

@saschagrunert -- I am curious about the reason why we tried to have the rc.0 cut one week after Test Freeze v1.22. 🤔 
What I can see is in v1.21 we coupled Test Freeze and rc.0 and in v1.23 we are proposing the same.

**Comment by reylejano** [comment]
Thank you @palnabarun
IMO, since there's only 6 days apart between beta.0 and rc.0 that we don't need a beta.1 but I defer to release branch management on this decision

Informal lgtm for now until there's more feedback to @palnabarun [questions](https://github.com/kubernetes/sig-release/pull/1688#issuecomment-914309295)

File: releases/release-1.23/README.md
Comment (saschagrunert) [inline releases/release-1.23/README.md:61]: This is already overdue, let's move it to Sep 14 to relax the pressure on our side.

**Review by saschagrunert — COMMENTED** [review]
> @reylejano @puerco @saschagrunert -- I've made the changes to alpha and beta according to what we discussed above.
> 
> I have not added a beta.1 in between beta.0 and rc.0 because I feel they are already 6 days apart and cutting a release in between may be tedious. What do you all think about it?

Agreed, let's stick to one beta.

> @saschagrunert -- I am curious about the reason why we tried to have the rc.0 cut one week after Test Freeze v1.22. 
> What I can see is in v1.21 we coupled Test Freeze and rc.0 and in v1.23 we are proposing the same.

I think the main reason was to not release on Thursdays, but the following Tuesday.

File: releases/release-1.23/README.md
Comment (palnabarun) [reply releases/release-1.23/README.md]: Ack. Done.

**Comment by palnabarun** [comment]
I have updated the deadlines based on the discussions above.

@reylejano @saschagrunert -- PTAL!

**Comment by reylejano** [comment]
/lgtm

@saschagrunert release the hold if it looks good to you

**Review by saschagrunert — APPROVED** [review]
/hold cancel

cc @Verolop

**Pull request merged**
